rmw_fastrtps_shared_cpp/src/qos_converter.hpp

```diff
@@ -0,0 +1,86 @@
+// Copyright 2016-2018 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef QOS_CONVERTER_HPP_
+#define QOS_CONVERTER_HPP_
+
+#include "rmw/types.h"
+
+#include "fastrtps/qos/WriterQos.h"
+#include "fastrtps/qos/ReaderQos.h"
+
+/*
+ * Converts the low-level QOS Policy; of type WriterQos or ReaderQos into rmw_qos_profile_t.
+ * Since WriterQos or ReaderQos does not have information about history and depth, these values are not set
+ * by this function.
+ *
+ * \param[in] dds_qos of type WriterQos or ReaderQos
+ * \param[out] qos the equivalent of the data in WriterQos or ReaderQos in rmw_qos_profile_t
+ */
+template<typename DDSQoSPolicyT>
+void
+dds_qos_policy_to_rmw_qos(
```

Comment:
Why a separated header?
It's not clear in which header the functions should live.

Also, I think that `dds_qos_to_rmw_qos` should be renamed to `dds_attributes_to_rmw_qos`, or something similar.
That would make a clearer difference between one function and the other.

Reply:
I put it in a different header to make the function package-private. Adding this as a new function inside `qos.hpp` would've made it public. Let me know if you think otherwise. 

I agree with the name `dds_attributes_to_rmw_qos`; makes more sense than what it is now. Will change it!

Reply:
I didn't realize it was a package private header, so it's ok.

I see you renamed this method to `dds_attributes_to_rmw_qos`:
https://github.com/ros2/rmw_fastrtps/blob/40f95239ada386b3fcb3abffec0a8691eec7c23a/rmw_fastrtps_shared_cpp/src/qos_converter.hpp#L34-L36
Though that one is transforming a `dds qos` into a `rmw qos` (and not a `dds publisher/subscriber attributes` into a `rmw qos`). So it should be named `dds_qos_to_rmw_qos` as before.

My idea was to rename this other to `dds_attribute_to_rmw_qos`:
https://github.com/ros2/rmw_fastrtps/blob/40f95239ada386b3fcb3abffec0a8691eec7c23a/rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/qos.hpp#L47-L51
As this one in the one transforming the `dds publisher/subscriber attributes` into a `rmw qos`.

Reply:
Ah! You're right; the excitement around getting this PR merged got the better of me! I will change it back to how it was before. 

As for renaming `dds_qos_to_rmw_qos` would you suggest that be done as a part of this PR?

Reply:
I find having both `dds_qos_to_rmw_qos` and `dds_qos_policies_to_rmw_qos` quite confusing (if I remember correctly, that was the original proposal).
I think it would be good to do the change as part of this PR.
It shouldn't be more than doing find and replace :wink: (through `rmw_fastrtps_cpp`, `rmw_fastrtps_dynamic_cpp` and `rmw_fastrtps_shared_cpp`).

Reply:
I see you have sent https://github.com/ros2/rmw_fastrtps/pull/330, which is ok.
Please, rename [this function](https://github.com/ros2/rmw_fastrtps/blob/40f95239ada386b3fcb3abffec0a8691eec7c23a/rmw_fastrtps_shared_cpp/src/qos_converter.hpp#L34-L36) back to `dds_qos_to_rmw_qos` and we're done.

Reply:
Yeah, I had the PR ready by the time your comment came in :P.

I've renamed the function back to `dds_qos_to_rmw_qos` and also made the necessary changes in the tests :)